list/src/lib.rs

```diff
@@ -0,0 +1,31 @@
+extern crate common;
+
+use common::args::Args;
+use common::first_run::check_first_run;
+use common::project_paths::{project_paths, CONFIG_EXTENSION};
+
+use std::path::PathBuf;
+
+pub fn exec(args: Args) -> Result<(), String> {
+    let project_paths = project_paths(&args);
+    check_first_run(&project_paths.project_directory)?;
+
+    let projects: Vec<String> = project_paths
+        .project_directory
+        .read_dir()
+        .map_err(|_| "Could not read the dir")?
+        .filter_map(|path| path.ok())
+        .map(|path| PathBuf::from(path.file_name()))
+        .filter_map(|buf| match buf.extension().and_then(|x| x.to_str()) {
+            Some(CONFIG_EXTENSION) => buf
+                .file_stem()
+                .and_then(|x| x.to_str())
+                .map(|x| x.to_string()),
+            _ => None,
+        })
+        .collect();
+
+    println!("{}", projects.join(" "));
```

Comment:
I also ended up using two tab characters here. `\t\t` which I found presented a bit nicer. I'm sure there's some kind of better standardized list spacing. Any thoughts?

Reply:
This seems like a great call! A space was just the simplest thing I could think of. I think using some kind of tabs makes sense so it lays out in a table style layout a bit nicer!